modules/hal_gigadevice/CMakeLists.txt

```diff
@@ -6,20 +6,32 @@ if(CONFIG_HAS_GD32_HAL)
 string(TOUPPER ${CONFIG_SOC} gd32_soc_uc)
 
 set(gd32_soc_dir     ${ZEPHYR_HAL_GIGADEVICE_MODULE_DIR}/${CONFIG_SOC_SERIES})
-set(gd32_cmsis_dir   ${gd32_soc_dir}/cmsis/gd/${CONFIG_SOC_SERIES})
+if(CONFIG_SOC_FAMILY_GD32_ARM)
+  set(gd32_soc_sys_dir   ${gd32_soc_dir}/cmsis/gd/${CONFIG_SOC_SERIES})
+elseif(CONFIG_SOC_SERIES_GD32VF103)
+  set(gd32_soc_sys_dir   ${gd32_soc_dir}/riscv)
+  zephyr_include_directories(${gd32_soc_dir}/riscv/drivers)
+endif()
+
 set(gd32_std_dir     ${gd32_soc_dir}/standard_peripheral)
 set(gd32_std_src_dir ${gd32_std_dir}/source)
 
 zephyr_library_named(hal_gigadevice)
 
 zephyr_compile_definitions(${gd32_soc_uc})
 
+if(${CONFIG_GD32_HXTAL_8MHZ})
+  zephyr_library_compile_definitions(HXTAL_VALUE=8000000)
+elseif(${CONFIG_GD32_HXTAL_25MHZ})
+  zephyr_library_compile_definitions(HXTAL_VALUE=25000000)
+endif()
```

Comment:
I think these definitions are required by HAL public headers, but `zephyr_library_compile_definitions` uses `PRIVATE` scope. See `gd32vf103/riscv/include/gd32vf103.h` in HAL. Looks like `HXTAL_VALUE_8M` and `HXTAL_VALUE_25M` also need to be defined.

Reply:
Thanks for reviewing.

I'll be correcting it.
I tried some ways to make this part private, but There was no good way. Some it was work fine but unnatural implementation.
I understood it is needed to define as public.

- - -
@tejlmand 

https://github.com/zephyrproject-rtos/zephyr/pull/34970#discussion_r767627166

I tried it...

Define HXTAL_VALUE in compiler option in hal_gigadevice module. And also define that in soc.h too.
It seems to be working fine. But it is a bit strange implementation.

Original GigaDevice Firmware uses with define HXTAL_VALUE in command-line for all compilations.
Because GigaDevice Firmware's header file references HXTAL_VALUE, Thus there is a possibility to reference that in all compilations.

Therefore, it must use zephyr_compile_definitions.

Reply:
> Therefore, it must use zephyr_compile_definitions.

So seems correct that this should go the `zephyr_compile_definitions()`, but as I wrote: https://github.com/zephyrproject-rtos/zephyr/pull/34970#discussion_r767627166
> Code might be correct, but without a comment here, or in the commit message I cannot judge if this is correct.

So please add a comment in the code why this is needed. From the comments above you seem to be aware why this is needed.
So please add a comment in the CMake code with this information.

Reply:
I got it now.
Adding more comments to CMakeLists.txt.